src/components/CheckIn/PiecesList.js

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import { FormattedMessage } from 'react-intl';
+import PropTypes from 'prop-types';
+
+import {
+  Checkbox,
+  MultiColumnList,
+} from '@folio/stripes/components';
+
+const PiecesList = ({ checkedItemsMap, items = [], toggleItem, toggleAll, isAllChecked = false }) => {
+  const resultsFormatter = {
+    'isChecked': piece => (
+      <Checkbox
+        checked={!!checkedItemsMap[piece.id]}
+        type="checkbox"
+        onChange={() => toggleItem(piece)}
+      />
+    ),
+    'title': piece => piece.title,
+    'piece': piece => piece.caption,
+    'supplement': piece => (
+      <Checkbox
+        checked={piece.supplement}
+        disabled
+        type="checkbox"
+      />
+    ),
+    'poLineNumber': piece => piece.poLineNumber,
+    'comment': piece => piece.comment,
+    'pieceStatus': piece => piece.receivingStatus,
+  };
+
+  return (
+    <MultiColumnList
+      contentData={items}
+      formatter={resultsFormatter}
+      visibleColumns={['isChecked', 'title', 'piece', 'supplement', 'poLineNumber', 'comment', 'pieceStatus']}
```

Comment:
ideally, it should be a constant to not create new array each render, but it doesn't make a lot sense because of  `columnMapping`

Reply:
sure, it's a good point, but in our case I think GC could handle this w/o huge impact on memory